edb/ir/scopetree.py

```diff
@@ -585,13 +585,15 @@ def find_unfenced(self, path_id: pathid.PathId) \
 
         return None, unnest_fence_seen
 
-    def find_by_unique_id(self, unique_id: int) \
-            -> Optional[ScopeTreeNode]:
+    def find_by_unique_id(self, unique_id: int) -> Optional[ScopeTreeNode]:
+        result: Optional[ScopeTreeNode] = None
+
         for node in self.descendants:
             if node.unique_id == unique_id:
-                return node
+                result = node
+                break
 
-        return None
+        return result
```

Comment:
Did you have to make those changes to make Mypy happy?

Reply:
Yes, otherwise it complained that I was returning `Any`

Reply:
That looks like a bug. It probably can't unwind the double generator right?

Reply:
Yeah, this was probably some intermediate issue.  It's not reproducing now, so I reverted back to the original formulation.